UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -276,24 +276,44 @@ private static MatrixInfo CreateMatrixInfoFromMatrix(Matrix matrix, int id)
 	}
 
 	/// Finds first matrix that is not empty at given world pos
-	public static MatrixInfo AtPoint(Vector3Int worldPos, bool isServer)
+	public static MatrixInfo AtPoint(Vector3Int worldPos, bool isServer, MatrixInfo possibleMatrix = null)
 	{
+		//for performance, this is just a suggestion on which matrix we believe this point could be in, not always correct
+		if (possibleMatrix is null == false && IsInMatrix(worldPos, isServer, possibleMatrix))
```

Comment:
```suggestion
		if (possibleMatrix != null && IsInMatrix(worldPos, isServer, possibleMatrix))
```

Lets just change it back